### PR TITLE
This PR cleans up several schemes that we would not want people to look at, use or benchmark.

### PR DESCRIPTION
**Pull request by lemire**
We have accumulated a lot of schemes that are unlikely to be useful. They add noise and confusion for people look at the code. 

It can be stressful when studying a code base to have all this dead code. How are you going to tell what is relevant and what is leftover from unsuccessful experiments? I think it is best to remove old code.

Note that with git, this code is not really gone: we can easily access it at any time by going to older commits.

This PR removes 4000 lines of code!

### CODE REVIEW

**Comment by lemire** [comment]
@thomasmueller I think we should consider this PR. Would you look at it?

Note that deleted schemes are always available under git, the history is not lost.

File: benchmarks/bulk-insert-and-query.cc [hunk 1]
```diff
@@ -394,28 +391,12 @@ int main(int argc, char * argv[]) {
...
-    {111, "XorBinaryFuse16-FSorted"},
-    {112, "XorBinaryFuse8-4Wise-PSorted"},
-    {113, "XorBinaryFuse16-4Wise-PSorted"},
-    {114, "XorBinaryFuse8-OneHash"},
-    {115, "XorBinaryFuse16-OneHash"},
-    {116, "XorBinaryFuse8"},
-    {117, "XorBinaryFuse16"},
-    {118, "XorBinaryFuse8-4Wise"},
-    {119, "XorBinaryFuse16-4Wise"},
+    {96, "XorBinaryFuse8 (Naive)"},
```
Comment (thomasmueller) [inline benchmarks/bulk-insert-and-query.cc:394]: I would actually prefer to not use spaces in the names... as that simplifies converting the output to e.g. Excel. So could we keep XorBinaryFuse8-Naive ?

File: benchmarks/bulk-insert-and-query.cc [hunk 2]
```diff
@@ -394,28 +391,12 @@ int main(int argc, char * argv[]) {
...
+    {97, "XorBinaryFuse16 (Naive)"},
+    {116, "XorBinaryFuse8-LowMem"},
```
Comment (thomasmueller) [inline benchmarks/bulk-insert-and-query.cc:396]: Not sure why you want to use the "LowMem" term here... sure it uses a bit less memory, but not a lot. Why use an extension at all? Isn't "XorBinaryFuse8" sufficient?

File: benchmarks/bulk-insert-and-query.cc [hunk 3]
```diff
@@ -344,11 +344,8 @@ int main(int argc, char * argv[]) {
   std::map<int,std::string> names = {
     // Xor
     {1000, "Xor8naive"}, {1002, "Xor16naive"},
-    {2000, "Xor8prefetch"}, {2002, "Xor16prefetch"},
     {0, "Xor8"}, {1, "Xor12"}, {2, "Xor16"},
     {3, "Xor+8"}, {4, "Xor+16"},
-    {5, "Xor10"}, {6, "Xor10.666"},
```
Comment (thomasmueller) [inline benchmarks/bulk-insert-and-query.cc:350]: I would actually keep them, because they have been referenced in the fuse filter paper (I think).

**Review by thomasmueller — COMMENTED** [review]
See my comments... yes most of those we can remove.

**Comment by lemire** [comment]
I'm fine with all your comments and I agree we can (and should) drop the lowmem qualifier.

The goal here is to make it easier for people who look at this code base to figure out what is what. We currently have way too many classes that should be ignored by readers (but they can't tell apart what they should ignore from what they should not).

**Pull request merged**